docs/source/pyopenms_in_r.rst

```diff
@@ -0,0 +1,243 @@
+
+.. code-block:: R
+
+    py_help(exp)
+    Help on MSExperiment object:
+
+    class MSExperiment(__builtin__.object)
+     |  Methods defined here:
+     ...
+     |  getNrChromatograms(...)
+     |      Cython signature: size_t getNrChromatograms()
+     |
+     |  getNrSpectra(...)
+     |      Cython signature: size_t getNrSpectra()
+     |
+     ...
+
+
+which indicates that the variable ``exp`` has (among others) the functions
+``getNrSpectra`` and ``getNrChromatograms``. We can now try these functions:
+
+.. code-block:: R
+
+    exp$getNrSpectra()
+    4
+    exp$getNrChromatograms()
+    2
+
+and indeed we see that we get information about the underlying MS data. We can
+iterate through the spectra as follows:
+
+Visualize spectra
+^^^^^^^^^^^^^^^^^
+
+You can easily visualise ms1 level precursor maps:
+
+.. code-block:: R
+
+    spectra = py_to_r(msexp$getSpectra())
+
+    ms1=sapply(spectra, function(x) x$getMSLevel()==1)
+    peaks=sapply(spectra[ms1], function(x) cbind(do.call("cbind", x$get_peaks()),x$getRT()))
+    peaks=do.call("rbind", peaks)
+    peaks_df=data.frame(peaks)
+    colnames(peaks_df)=c('MZ','Intensity','RT')
+    peaks_df$Intensity=log10(peaks_df$Intensity)
+
+    ggplot(peaks_df, aes(x=RT, y=MZ) ) +
+    geom_point(size=1, aes(colour = Intensity), alpha=0.25) +
+    theme_minimal() +
+    scale_colour_gradient(low = "blue", high = "yellow")
+
+    (plot)
+
+Or visualize a particular ms2 spectrum:
+
+.. code-block:: R
+
+    spectra = py_to_r(msexp$getSpectra())
+
+    ms2=spectra[!ms1][[1]]$get_peaks()
+    peaks_ms2=do.call("cbind", ms2)
+    peaks_ms2=data.frame(peaks_ms2)
+
+    ggplot(peaks_ms2, aes(x=X1, y=X2)) +
+    geom_segment( aes(x=X1, xend=X1, y=0, yend=X2)) +
+    geom_segment( aes(x=X1, xend=X1, y=0, yend=-X2)) + 
+    theme_minimal()
+
+    (plot)
+
+Iteration
+^^^^^^^^^
+
+Iterating over pyopenmsobjects is not equal to iterating over R vectors or lists.
+
+Therefore we can not directly apply the usual functions such as apply() and have to use reticulate::iterate() instead:
+
+.. code-block:: R
+
+    spectrum = ropenms$MSSpectrum()
+    mz = seq(1500, 500, -100)
+    i = seq(10, 2000, length.out = length(mz))
+    spectrum$set_peaks(list(mz, i))
+
+    iterate(spectrum, function(x) {print(paste0("M/z :" , x$getMZ(), " Intensity: ", x$getIntensity()))})
+
+    [1] "M/z :1500.0 Intensity: 10.0"
+    [1] "M/z :1400.0 Intensity: 209.0"
+    [1] "M/z :1300.0 Intensity: 408.0"
+    [1] "M/z :1200.0 Intensity: 607.0"
+    [1] "M/z :1100.0 Intensity: 806.0"
+    [1] "M/z :1000.0 Intensity: 1005.0"
+    [1] "M/z :900.0 Intensity: 1204.0"
+    [1] "M/z :800.0 Intensity: 1403.0"
+    [1] "M/z :700.0 Intensity: 1602.0"
+    [1] "M/z :600.0 Intensity: 1801.0"
+    [1] "M/z :500.0 Intensity: 2000.0"
+
+or as a way around:
+
+.. code-block:: R
+
+    for (i in seq(0,spectrum$size()-1)) {
+          print(spectrum[i]$getMZ())
+          print(spectrum[i]$getIntensity())
+    }
```

Comment:
this actually fails for me:

```
> for (i in seq(0,spectrum$size()-1))
+ {
+       print(spectrum[i]$getMZ())
+       print(spectrum[i]$getIntensity())
+ }
Error in spectrum$size() - 1 : non-numeric argument to binary operator
> 

```

Reply:
this one worked for me previously when I did the reticulate import without the convert=FALSE option.

ropenms=import("pyopenms")

However then I came across a problem converting 64bit Integers. So I guess it's better to import it without conversion and do the necessary conversions manually.

py_to_r(spectrum$size())

Reply:
> However then I came across a problem converting 64bit Integers.

what are the issues here? It seems like it would make our lives much easier if we can use `convert = TRUE`

Reply:
So we noticed when loading a featureXML file, that we could not access the respective ID of a feature as these IDs are 64bit large eg. f_4751408525657442109.

reticulate was then giving -1 as output for $getUniqueId()

setting the automatic conversion to FALSE seemed to be a way to fix it, but possilby there is another solution to it as well.

Reply:
sounds related to https://github.com/rstudio/reticulate/issues/323